core/DataTable/Filter/Sort.php

```diff
@@ -80,20 +83,20 @@ public function numberSort($rowA, $rowB)
             } else {
                 return -1 * $this->sign * strnatcasecmp($rowA[1], $rowB[1]);
             }
-        } elseif (!isset($rowB[0])) {
-            return -1;
+        } elseif (!isset($rowB[0]) && !isset($rowA[0])) {
+            return -1 * $this->sign * strnatcasecmp($rowA[1], $rowB[1]);
         } elseif (!isset($rowA[0])) {
```

Comment:
If both values (`$rowA[0]` and `$rowB[0]`) were not set, it actually never reached the `return 0` and `return 1` statement. This was a bug. Also if both values are not set, we do now sort by label which is a new feature.